### PR TITLE
First Contact

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
rscadd: A new mode is available through the mapswitcher: First Contact.
/:cl:

### CODE REVIEW

File: code/game/machinery/floodlight.dm [hunk 1]
```diff
@@ -13,7 +13,8 @@
 	var/brightness_on = 8		//can't remember what the maxed out value is
 
 /obj/machinery/floodlight/New()
-	cell = new/obj/item/weapon/cell/crap(src)
+	if(!cell)
+		cell = new/obj/item/weapon/cell/crap(src)
```
Comment (caelaislinn) [inline code/game/machinery/floodlight.dm]: This seems redundant, what could have created a cell before New() is called?

File: code/modules/halo/languages.dm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-
+c
```
Comment (caelaislinn) [inline code/modules/halo/languages.dm]: The return of c

File: code/modules/halo/misc/capture_node_mobs.dm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	possible_weapons = list(/obj/item/weapon/gun/energy/plasmapistol, /obj/item/weapon/gun/projectile/needler)
+
+/mob/living/simple_animal/hostile/defender_mob/cov/kig
+	name = "Covenant Defender (Kig Yar)"
+	health = 100
+	resistance = 20
+	icon_state = "kigyar"
+	icon_living = "kigyar"
+	icon_dead = "dead_kigyar"
+	possible_weapons = list(/obj/item/weapon/gun/energy/plasmapistol, /obj/item/weapon/gun/projectile/type51carbine, /obj/item/weapon/gun/projectile/type31needlerifle)
```
Comment (caelaislinn) [inline code/modules/halo/misc/capture_node_mobs.dm]: I moved these mob definitions into their faction folders, are you moving them back or redefining them here?

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 4]
```diff
@@ -135,29 +135,30 @@
...
-					v.visible_message("<span class = 'notice'>[name] enters [v.name]'s gunner seat.</span>")
-					return 1
-	else
-		if(leader_follow && loc != leader_follow.loc)
-			set_leader(null)
-		walk(src,0)
+		else
+			if(leader_follow && loc != leader_follow.loc)
+				set_leader(null)
+			walk(src,0)
```
Comment (caelaislinn) [inline code/modules/mob/living/simple_animal/simple_animal.dm:161]: I've touched this proc in #2404 to make it work with assault pathing, but I can resolve that conflict if it arises

File: code/modules/projectiles/projectile.dm [hunk 5]
```diff
@@ -197,6 +197,9 @@
 		return 0
 
 	//hit messages
+	if(isnull(target_mob))
```
Comment (caelaislinn) [inline code/modules/projectiles/projectile.dm:200]: Why would target_mob be null at this point?

File: code/game/machinery/floodlight.dm (hunk 1 above)
Comment (X0-11) [reply code/game/machinery/floodlight.dm]: This is so I can override this in subtypes to create better capacity cells.

File: code/modules/halo/languages.dm (hunk 2 above)
Comment (X0-11) [reply code/modules/halo/languages.dm]: c

File: code/modules/halo/misc/capture_node_mobs.dm (hunk 3 above)
Comment (X0-11) [reply code/modules/halo/misc/capture_node_mobs.dm]: uh, I think that's an incorrect merge thing

File: code/modules/projectiles/projectile.dm (hunk 5 above)
Comment (X0-11) [reply code/modules/projectiles/projectile.dm:200]: Dying NPCs creating corpses and deleting themselves, I think it was.

File: code/game/machinery/floodlight.dm (hunk 1 above)
Comment (caelaislinn) [reply code/game/machinery/floodlight.dm]: The cell type should be stored as a variable, that way you only need to override New() once, in the top level ancestor

**Review by caelaislinn — APPROVED** [review]
Waiting on travis pass to merge

**Pull request merged**